transport_nantes/mailing_list/views.py

```diff
@@ -299,3 +300,61 @@ def get_context_data(self, **kwargs):
             context["mailing_list"] = get_object_or_404(
                 MailingList, id=mailing_list_id)
         return context
+
+
+class MailingListUnsubscribeWithToken(FormView):
+    form_class = SubscribeUpdateForm
+    success_url = reverse_lazy('mailing_list:unsubscribe_finish')
+    template_name = 'mailing_list/validate_form.html'
+
+    def get(self, *args, **kwargs):
+        token = self.kwargs['token']
+        if not token:
+            logger.info("Try to unbscrucribe to mailist without token.")
+            return HttpResponseNotFound(
+                "<h1>Désolé, la page que vous recherchez est introuvable</h1>")
+        try:
+            id_user,  the_tb_email_send_record_id, persistent = token_valid(
+                token, mailing=True)
+        except ValueError:
+            logger.info("Invalid token when unbscrucribe to mailist.")
+            return HttpResponseNotFound(
+                "<h1>Désolé, la page que vous recherchez est introuvable</h1>")
+        if (id_user is None or the_tb_email_send_record_id is None or
+                persistent == 0):
+            logger.info("Invalid token when unbscrucribe to mailist.")
+            return HttpResponseNotFound(
+                "<h1>Désolé, la page que vous recherchez est introuvable</h1>")
+        return super().get(*args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        token = self.kwargs['token']
+        email, the_tb_email_send_record_id, persistent = token_valid(
+            token, mailing=True)
+        context['user'] = get_object_or_404(User, email=email)
+        context['tb_email_send_record'] = get_object_or_404(
+            TopicBlogEmailSendRecord, id=the_tb_email_send_record_id)
+        context['mailing_list'] = context['tb_email_send_record'].mailinglist
+        # Check the current state and set context if the user is unsub
+        current_state = user_current_state(
+            context['user'], context['mailing_list'])
+        if current_state.event_type == "unsub":
+            context['is_unsub'] = True
+        context['is_from_token'] = True
```

Comment:
What's the point of `context['is_from_token'] = True` ? If this is always True, why does it exist anyway ?

Reply:
I use same template for status and token i need to know where the request is from for the render.

Reply:
Could you please add a comment to explain this please ? It's not immediately obvious :slightly_smiling_face: